keras_cv/layers/preprocessing/fmix.py

```diff
@@ -0,0 +1,221 @@
+
+        return scale * param
+
+    def sample_mask_from_transform(self, decay, shape, ch=1):
+        # Sampling low frequency map from fourier transform.
+
+        freqs = self.apply_fftfreq(shape[0], shape[1])
+        spectrum = self.get_spectrum(freqs, decay, ch, shape[0], shape[1])
+        spectrum = tf.complex(spectrum[:, 0], spectrum[:, 1])
+
+        mask = tf.math.real(tf.signal.irfft2d(spectrum, shape))
+        mask = mask[:1, : shape[0], : shape[1]]
+
+        mask = mask - tf.reduce_min(mask)
+        mask = mask / tf.reduce_max(mask)
+        return mask
+
+    def binarise_mask(self, mask, lam, in_shape, edge_softening=0.0):
+        # Create the final mask from the sampled values.
+
+        idx = tf.argsort(tf.reshape(mask, [-1]), direction="DESCENDING")
+        mask = tf.reshape(mask, [-1])
+        num = tf.math.round(lam * tf.cast(tf.size(mask), tf.float32))
+
+        eff_soft = edge_softening
+
+        max_compare = tf.math.maximum(lam, (1 - lam))
+        min_compare = tf.math.minimum(lam, (1 - lam))
+        eff_soft = tf.where(edge_softening > max_compare, max_compare, edge_softening)
+        eff_soft = tf.where(edge_softening > min_compare, min_compare, edge_softening)
+
+        soft = tf.cast(tf.size(mask), tf.float32) * eff_soft
+        num_low = tf.cast(num - soft, tf.int32)
+        num_high = tf.cast(num + soft, tf.int32)
+
+        updates = tf.concat(
+            [
+                tf.ones((num_low,), tf.float32),
+                tf.cast(tf.linspace(1, 0, num_high - num_low), tf.float32),
+                tf.zeros((tf.size(mask) - num_high,), tf.float32),
+            ],
+            0,
+        )
+
+        mask = tf.scatter_nd(
+            tf.expand_dims(idx, -1), updates, tf.expand_dims(tf.size(mask), -1)
+        )
+
+        mask = tf.reshape(mask, in_shape)
+        return mask
+
+    def _batch_augment(self, inputs):
+        images = inputs.get("images", None)
+        labels = inputs.get("labels", None)
+        if images is None or labels is None:
+            raise ValueError(
+                "FMix expects inputs in a dictionary with format "
+                '{"images": images, "labels": labels}.'
+                f"Got: inputs = {inputs}"
+            )
+        images, lambda_sample, permutation_order = self._fmix(images)
+        if labels is not None:
+            labels = self._update_labels(labels, lambda_sample, permutation_order)
+            inputs["labels"] = labels
+        inputs["images"] = images
+        return inputs
+
+    def _augment(self, inputs):
+        raise ValueError(
+            "FMix received a single image to `call`.  The layer relies on "
+            "combining multiple examples, and as such will not behave as "
+            "expected.  Please call the layer with 2 or more samples."
+        )
+
+    def _fmix(self, images):
+        shape = tf.shape(images)
+        permutation_order = tf.random.shuffle(tf.range(0, shape[0]), seed=self.seed)
```

Comment:
@quantumalaviya In order to reproduce the o/p with a given seed value, I think it's needed to do as follows:

```python
tf.random.set_seed(seed=self.seed)
tf.random.shuffle(tf.range(0, shape[0]), seed=self.seed)
```
Can u please check?

Reply:
The `BaseImageAugmentationLayer` takes care of this, I believe. It inherits the `BaseRandomLayer` to which the seed is passed.

Reply:
@quantumalaviya I think that to use that `seed` you would need to replace `tf.random` ops with `self._random_generator`.

Looking at the code of [BaseRandomLayer](https://github.com/keras-team/keras/blob/275a0707c066bd90501e3fb3a0c26b1b676c06d3/keras/engine/base_layer.py#L3342)
```python
self._random_generator = backend.RandomGenerator(
        seed, force_generator=force_generator)
```

Reply:
Yes, that actually makes sense @sebastian-sz. I didn't realize I haven't used the `_random_generator` everywhere.

Correct me if I am wrong here but `backend.RandomGenerator` doesn't provide a shuffle function (although it claims it should be used for shuffling). Am I missing something here?

Reply:
@quantumalaviya Yes you are right - this particular op does not seem to have Generator equivalent.

Reply:
I believe @innat's approach should work then:

```python
tf.random.set_seed(seed=self.seed)
tf.random.shuffle(tf.range(0, shape[0]), seed=self.seed)
```

Reply:
I don't think we want to be setting a seed inside of call like this actually.

Reply:
For now, feel free to just pass `seed` to the `tf.random.shuffle` as you have done.